packages/StaticTypeMapper/StaticTypeMapper.php

```diff
@@ -112,4 +115,9 @@ public function mapPHPStanPhpDocTypeNodeToPHPStanType(TypeNode $typeNode, Node $
         $nameScope = $this->nameScopeFactory->createNameScopeFromNode($node);
         return $this->phpDocTypeMapper->mapToPHPStanType($typeNode, $node, $nameScope);
     }
+
+    public function mapIdentifierTypeNodeToPHPStanType(IdentifierTypeNode $typeNode, Node $node): Type
```

Comment:
the idea behind this method is: 

- we see in https://github.com/rectorphp/rector/issues/8077#issuecomment-1645081188 that name scope creation is slow
- IdentifierTypeMapper does not need the name-scope for its inner workings -> lets try a separate fast path in case we know the `TypeNode` so we can skip name scope creation